pkg/security/admission/admission.go

```diff
@@ -349,6 +384,91 @@ func getPreallocatedLevel(ns *kapi.Namespace) (string, error) {
 	return level, nil
 }
 
+// getSupplementalGroupsAnnotation provides a backwards compatible way to get supplemental groups
+// annotations from a namespace by looking for SupplementalGroupsAnnotation and falling back to
+// UIDRangeAnnotation if it is not found.
+func getSupplementalGroupsAnnotation(ns *kapi.Namespace) (string, error) {
+	groups, ok := ns.Annotations[allocator.SupplementalGroupsAnnotation]
+	if ok {
+		return groups, nil
+	}
+	glog.V(4).Infof("unable to find supplemental group annotation %s falling back to %s", allocator.SupplementalGroupsAnnotation, allocator.UIDRangeAnnotation)
+
+	groups, ok = ns.Annotations[allocator.UIDRangeAnnotation]
+	if ok {
+		return groups, nil
+	}
+
+	return "", fmt.Errorf("unable to find supplemental group or uid annotation for namespace %s", ns.Name)
+}
+
+// getPreallocatedFSGroup gets the annotated value from the namespace.
+func getPreallocatedFSGroup(ns *kapi.Namespace) ([]kapi.IDRange, error) {
+	groups, err := getSupplementalGroupsAnnotation(ns)
+	if err != nil {
+		return nil, err
+	}
+	if len(groups) == 0 {
+		return nil, fmt.Errorf("found annotation %s but it was empty", allocator.SupplementalGroupsAnnotation)
+	}
+	glog.V(4).Infof("got preallocated value for groups: %s in namespace %s", groups, ns.Name)
+
+	blocks, err := parseSupplementalGroupAnnotation(groups)
+	if err != nil {
+		return nil, err
+	}
+	return []kapi.IDRange{
+		{
+			Min: int64(blocks[0].Start),
+			Max: int64(blocks[0].Start),
+		},
+	}, nil
+}
+
+// getPreallocatedSupplementalGroups gets the annotated value from the namespace.
+func getPreallocatedSupplementalGroups(ns *kapi.Namespace) ([]kapi.IDRange, error) {
+	groups, err := getSupplementalGroupsAnnotation(ns)
+	if err != nil {
+		return nil, err
+	}
+	if len(groups) == 0 {
+		return nil, fmt.Errorf("found annotation %s but it was empty", allocator.SupplementalGroupsAnnotation)
+	}
+	glog.V(4).Infof("got preallocated value for groups: %s in namespace %s", groups, ns.Name)
+
+	blocks, err := parseSupplementalGroupAnnotation(groups)
+	if err != nil {
```

Comment:
When using a strategy that requires values you'll get an error creating the strategy if no values are given.  This allows us to indicate that the error was in parsing the annotation rather than a generic "bad strategy" message.